src/plugins/intel_cpu/src/nodes/non_max_suppression.cpp

```diff
@@ -188,25 +149,53 @@ void NonMaxSuppression::initSupportedPrimitiveDescriptors() {
         outDataConf.emplace_back(LayoutType::ncsp, outPrecision);
     }
 
-    // as only FP32 and ncsp is supported, and kernel is shape agnostic, we can create here. There is no need to recompilation.
+    impl_desc_type impl_type = impl_desc_type::ref;
+
+#if defined(OPENVINO_ARCH_X86_64)
+    using namespace dnnl::impl::cpu;
+
+    // As only FP32 and ncsp is supported, and kernel is shape agnostic, we can create here. There is no need to recompilation.
     createJitKernel();
 
     x64::cpu_isa_t actual_isa = x64::isa_undef;
     if (m_jit_kernel) {
         actual_isa = m_jit_kernel->getIsa();
     }
-
-    impl_desc_type impl_type;
     switch (actual_isa) {
         case x64::avx512_core: impl_type = impl_desc_type::jit_avx512; break;
         case x64::avx2:        impl_type = impl_desc_type::jit_avx2;   break;
         case x64::sse41:       impl_type = impl_desc_type::jit_sse42;  break;
         default:               impl_type = impl_desc_type::ref;
     }
+#endif // OPENVINO_ARCH_X86_64
 
     addSupportedPrimDesc(inDataConf, outDataConf, impl_type);
 }
 
+void NonMaxSuppression::createPrimitive() {
+    const auto inputs_num = inputShapes.size();
+
+    if (inputs_num > NMS_MAX_OUTPUT_BOXES_PER_CLASS && m_const_inputs[NMS_MAX_OUTPUT_BOXES_PER_CLASS]) {
```

Comment:
Could you please elaborate why do you move this code from constructor to `createPrimitive`?

Reply:
That looks safer. This fn is called after all plugin optimizations and we sure that constants have node-specific precisions which were set in fn `initSupportedPrimitiveDescriptors`